pkg/openstack/loadbalancer.go

```diff
@@ -2390,7 +2390,7 @@ func (lbaas *LbaasV2) deleteLoadBalancer(loadbalancer *loadbalancers.LoadBalance
 					Protocol: proto,
 					Port:     int(port.Port),
 				}]
-				if isPresent && cpoutil.Contains(listener.Tags, loadbalancer.Name) {
+				if isPresent && cpoutil.Contains(listener.Tags, svcConf.lbName) {
```

Comment:
Can you explain why this is different to loadbalancer.Name? I've read this function and the calling function, in which you're now setting svcConf.lbName to the return value of `lbaas.GetLoadBalancerName`. Does this imply that the loadbalancer name (retrieved by id?) is different to `GetLoadBalancerName`.

Reply:
Yes, so this all boils down to the support for shared LBs that I believe is implemented in a way that is very problematic codewise.

How does it work? You add an annotation to the Service that features the LB ID you want that Service use. The assumption is that the LB exists already and CPO will just add listener, pool and members (members should be the same really[^1]). The _secondary_ LB resources will be tagged with the LB name **that would be used for the secondary Service** so that we can distinguish them later on.

The bug here is that we looked up listeners comparing their tags with the name of the LB taken from OpenStack, which is based on the **primary** Service name. In that case we should use the secondary service name and this is the fix.

[^1]: Now this makes me think that we might have a bug that 2 Services sharing an LB can fight for their view of LB members. Fixed on reconciliation, but not great.

Reply:
This feels very fragile, but fixing that is definitely outside of the scope of this PR.

/lgtm

Reply:
It totally is and you'll probably be happy to see it disabled by default in OpenShift: openshift/cluster-cloud-controller-manager-operator#263.